test/integration/features/step_definitions/common-steps.js

```diff
@@ -0,0 +1,50 @@
+import {resolve} from 'path';
+import {promises as fs} from 'fs';
+import {After, When} from '@cucumber/cucumber';
+import stubbedFs from 'mock-fs';
+
+const packagePreviewDirectory = '../__package_previews__/mdx-deck';
+const stubbedNodeModules = stubbedFs.load(resolve(__dirname, '../../../../', 'node_modules'));
+
+After(function () {
+  stubbedFs.restore();
+});
+
+When('the project is scaffolded', async function () {
+  // eslint-disable-next-line import/no-extraneous-dependencies,import/no-unresolved
+  const mdxDeck = require('@form8ion/mdx-deck');
+  const {scaffold} = mdxDeck;
+
+  stubbedFs({
+    node_modules: stubbedNodeModules,
+    [packagePreviewDirectory]: {
```

Comment:
i really wish this wasnt all necessary. i've noticed that reading the template files has to be stubbed differently than the normal node_modules, but wish it could be done relative to the normal project root. i dont know the solution, if there is one, so moving forward with this for now

Reply:
I haven't worked through enough examples to know, but is there a consistent pattern that we could pull out a `package-preview-dep-stubber` test helper of some sort? Or are they all one-off? The thing that makes me lean in that direction is really what you're doing is not gross at all.. it's just the fact that you have to fiddle with all the pathing, etc. each time and manually.

Reply:
potentially. if we continue to fight it, it may be worth investing some time in.

the worst part is the version number being hard coded in the path. it prevents updates to that dependency w/o updating that path. as one of the first steps, it might be worth pulling that version number from the `package.json` at least, in order to decouple that a bit.